test/Dialect/Calyx/errors.mlir

```diff
@@ -1050,3 +1050,99 @@ module attributes {calyx.entrypoint = "main"} {
+    %r.in, %r.write_en, %r.clk, %r.reset, %r.out, %r.done = calyx.register @r : i32, i1, i1, i1, i32, i1
+    calyx.wires {
+
+    }
+    calyx.control {
+      // expected-error @+1 {{'calyx.invoke' op the input for 'calyx.invoke' is empty.}}
+      calyx.invoke@r() -> ()
+    }
+  }
+}
+
+// ----- 
+
+module attributes {calyx.entrypoint = "main"} {
+  calyx.component @main(%go: i1 {go}, %clk: i1 {clk}, %reset: i1 {reset}) -> (%done: i1 {done}) {
+    %c10 = hw.constant 10 : i32
+    %r.in, %r.write_en, %r.clk, %r.reset, %r.out, %r.done = calyx.register @r : i32, i1, i1, i1, i32, i1 
+    calyx.wires {
+
+    }
+    calyx.control {
+      // expected-error @+1 {{'calyx.invoke' op has a cell port as the destination with the incorrect direction.}}
+      calyx.invoke@r(%r.out = %c10) -> (i32)
+    }
+  }
+}
+
+// ----- 
+
+module attributes {calyx.entrypoint = "main"} {
+  calyx.component @main(%go: i1 {go}, %clk: i1 {clk}, %reset: i1 {reset}) -> (%done: i1 {done}) {
+    %c1 = hw.constant 1 : i1
+    %r.in, %r.write_en, %r.clk, %r.reset, %r.out, %r.done = calyx.register @r : i32, i1, i1, i1, i32, i1 
+    calyx.wires {
+
+    }
+    calyx.control {
+      // expected-error @+1 {{'calyx.invoke' op the go port of 'r' cannot appear here.}}
+      calyx.invoke@r(%r.write_en = %c1) -> (i1)
+    }
+  }
+}
+
+// -----
+
+module attributes {calyx.entrypoint = "main"} {
+  calyx.component @main(%go: i1 {go}, %clk: i1 {clk}, %reset: i1 {reset}) -> (%done: i1 {done}) {
+    %c1 = hw.constant 1 : i1
+    %r.in, %r.write_en, %r.clk, %r.reset, %r.out, %r.done = calyx.register @r : i32, i1, i1, i1, i32, i1 
+    calyx.wires {
+
+    }
+    calyx.control {
+      // expected-error @+1 {{'calyx.invoke' op the done port of 'r' cannot appear here.}}
+      calyx.invoke@r(%done = %r.done) -> (i1)
+    }
+  }
+}
+
+// ----- 
+
+module attributes {calyx.entrypoint = "main"} {
+  calyx.component @main(%go: i1 {go}, %clk: i1 {clk}, %reset: i1 {reset}) -> (%done: i1 {done}) {
+    %c10 = hw.constant 10 : i32
+    %add.left, %add.right, %add.out = calyx.std_add @add : i32, i32, i32
+    calyx.wires {
+
+    }
+    calyx.control {
+      // expected-error @+1 {{'calyx.invoke' op  'add' must have single go port and single done port, the 'add' has 0 go port and 0 done port.}}
+      calyx.invoke@add(%add.left = %c10, %add.right = %c10) -> (i32, i32)
+    }
+  }
+}
+
+// -----
+
+module attributes {calyx.entrypoint = "main"} {
+  calyx.component @main(%go: i1 {go}, %clk: i1 {clk}, %reset: i1 {reset}, %in : i32) -> (%done: i1 {done}) {
+    %c10 = hw.constant 10 : i32
+    %r0.in, %r0.write_en, %r0.clk, %r0.reset, %r0.out, %r0.done = calyx.register @r0 : i32, i1, i1, i1, i32, i1 
+    %r1.in, %r1.write_en, %r1.clk, %r1.reset, %r1.out, %r1.done = calyx.register @r1 : i32, i1, i1, i1, i32, i1
+    calyx.wires {
+      
+    }
+    calyx.control {
+      // expected-error @+1 {{'calyx.invoke' op all connections should involve the port of the invoke instance.}}
```

Comment:
What is this error message signaling? That the invoke should provide all ports of the invoked component? In general, that is not true; only the `go` and `done` ports of an invoke are needed.

Reply:
> What is this error message signaling? That the invoke should provide all ports of the invoked component? In general, that is not true; only the `go` and `done` ports of an invoke are needed.

What you can see is that we are invoking r0, but the port used here is r1.in.